libraries/helpers_nftables.rb

```diff
@@ -0,0 +1,112 @@
+module FirewallCookbook
+  module Helpers
+    module Nftables
+      include FirewallCookbook::Helpers
+      include Chef::Mixin::ShellOut
+
+      unless defined? CHAIN
+        CHAIN = {
+          in: 'INPUT',
+          out: 'OUTPUT',
+          pre: 'PREROUTING',
+          post: 'POSTROUTING',
+          # nil => "FORWARD",
+        }.freeze
+      end
+
+      unless defined? TARGET
+        TARGET = {
+          allow: 'accept',
+          reject: 'reject',
+          deny: 'drop',
+          masquerade: 'masquerade',
+          redirect: 'redirect',
+          log: 'log prefix "nftables:" group 0',
+        }.freeze
+      end
+
+      def build_firewall_rule(rule_resource)
+        return rule_resource.raw.strip if rule_resource.raw
+
+        ip = ipv6_rule?(rule_resource) ? 'ip6' : 'ip'
+        table = if [:pre, :post].include?(rule_resource.direction)
+                  'nat'
+                else
+                  'filter'
+                end
+        firewall_rule = if table == 'nat'
+                          "add rule #{ip} #{table} "
+                        else
+                          "add rule inet #{table} "
+                        end
+        firewall_rule << "#{CHAIN.fetch(rule_resource.direction.to_sym, 'FORWARD')} "
+
+        firewall_rule << "iif #{rule_resource.interface} " if rule_resource.interface
+        firewall_rule << "oif #{rule_resource.dest_interface} " if rule_resource.dest_interface
+
+        if rule_resource.source
+          source_with_mask = ip_with_mask(rule_resource, rule_resource.source)
+          Chef::Log.fatal source_with_mask
```

Comment:
Is this definetly the right log level? Seems strange to use `fatal` if it's not conditional

Reply:
Also needs more context around it, if I see this in the log I don't get why it is fatal, what did I do wrong?

Reply:
I did some debugging there and forgot to remove that line.  There should be no log entry at all, I will fix this.